src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -27,123 +26,38 @@ namespace Microsoft.Identity.Client.Http
     internal class HttpManager : IHttpManager
     {
         protected readonly IMsalHttpClientFactory _httpClientFactory;
+        private readonly Func<HttpResponse, bool> _retryCondition;
         public long LastRequestDurationInMs { get; private set; }
 
-        public HttpManager(IMsalHttpClientFactory httpClientFactory)
-        {
-            _httpClientFactory = httpClientFactory ?? 
-                throw new ArgumentNullException(nameof(httpClientFactory));
-        }
-
-        protected virtual HttpClient GetHttpClient()
-        {
-            return _httpClientFactory.GetHttpClient();
-        }
-
-        public async Task<HttpResponse> SendPostAsync(
-            Uri endpoint,
-            IDictionary<string, string> headers,
-            IDictionary<string, string> bodyParameters,
-            ILoggerAdapter logger,
-            CancellationToken cancellationToken = default)
-        {
-            HttpContent body = bodyParameters == null ? null : new FormUrlEncodedContent(bodyParameters);
-            return await SendPostAsync(endpoint, headers, body, logger, cancellationToken).ConfigureAwait(false);
-        }
-
-        public virtual Task<HttpResponse> SendPostAsync(
-            Uri endpoint,
-            IDictionary<string, string> headers,
-            HttpContent body,
-            ILoggerAdapter logger,
-            CancellationToken cancellationToken = default)
-        {
-            return SendRequestAsync(endpoint, headers, body, HttpMethod.Post, logger, cancellationToken: cancellationToken);
-        }
-
-        public virtual Task<HttpResponse> SendGetAsync(
-            Uri endpoint,
-            IDictionary<string, string> headers,
-            ILoggerAdapter logger,
-            bool retry = true,
-            CancellationToken cancellationToken = default)
-        {
-            return SendRequestAsync(endpoint, headers, null, HttpMethod.Get, logger, cancellationToken: cancellationToken);
-        }
-
         /// <summary>
-        /// Performs the GET request just like <see cref="SendGetAsync(Uri, IDictionary{string, string}, ILoggerAdapter, bool, CancellationToken)"/>
-        /// but does not throw a ServiceUnavailable service exception. Instead, it returns the <see cref="HttpResponse"/> associated
-        /// with the request.
+        /// A new instance of the HTTP manager with a retry *condition*. The retry policy hardcodes: 
+        /// - the number of retries (1)
+        /// - the delay between retries (1 second)
         /// </summary>
```

Comment:
Is this used for all requests, or just the initial request?

In Azure.Identity, we only reduce the default retry behavior for scenarios like DefaultAzureCredential where we want to fail fast. For scenarios where IMDS is intended explicitly, we allow default timeouts and retries.